File: lab-2/c++/StringS.cpp
#include "StringS.h"

using namespace std;

StringManipulator::StringManipulator() : text("") {}

StringManipulator::StringManipulator(const string& str) : text(str) {}

StringManipulator::StringManipulator(const StringManipulator& other) : text(other.text) {}

StringManipulator::~StringManipulator() {}

int StringManipulator::calculateLength() const {
    return text.length();
}

void StringManipulator::shiftRight() {
    if (!text.empty()) {
        char lastChar = text.back();
        text.pop_back();
        text.insert(text.begin(), lastChar);
    }
}

string StringManipulator::getText() const {
    return text;
}

StringManipulator& StringManipulator::operator=(const StringManipulator& other) {
    if (this != &other) {
        text = other.text;
    }
    return *this;
}

File: lab-1.2/c#/StringS.cs
using System;

public class StringManipulator
{
    private string text;

    public StringManipulator()
    {
        text = "";
    }

    public StringManipulator(string str)
    {
        text = str;
    }

    public StringManipulator(StringManipulator other)
    {
        text = other.text;
    }

    public int CalculateLength()
    {
        return text.Length;
    }

    public void ShiftRight()
    {
        if (!string.IsNullOrEmpty(text))
        {
            char lastChar = text[text.Length - 1];
            text = lastChar + text.Substring(0, text.Length - 1);
        }
    }

    public string GetText()
    {
        return text;
    }
}